core/services/head_tracker.go

```diff
@@ -340,6 +252,9 @@ func (ht *HeadTracker) listenForNewHeads() {
 			break
 		} else if err != nil {
 			ht.logger().Errorw("Error in new head subscription, unsubscribed", "err", err)
+			// subscribe() will call unsubscribeFromHead() which will panic
+			// closing the channel if already closed.
+			ht.headers = make(chan *models.Head)
```

Comment:
hm so it was being double-closed? Or perhaps was panicking on nil? Not sure if the comment is explanatory

In `unsubscribeFromHead` we also have 
```

	if !hl.connected {
		return nil
	}
```

so it should not attempt to close the channel if we are not connected. Unless the channel can be closed by the geth client? That would be similar to what you have in the test